### PR TITLE
Add tag "slow" if span is considered slow

### PR DESCRIPTION
**Pull request by quaff**
I'm using jaeger-ui, It can not query traces by <strong>span</strong> duration, but we could query traces by "slow=true" tag.

### CODE REVIEW

**Comment by pavolloffay** [comment]
@quaff jaeger can query spans by duration.

**Comment by quaff** [comment]
> @quaff jaeger can query spans by duration.


Duration is whole trace duration not span duration.

**Comment by yurishkuro** [comment]
duration is for the selected service/operation, it applies anywhere in the trace

**Comment by quaff** [comment]
> duration is for the selected service/operation, it applies anywhere in the trace

There are many operations including `Query` `Update` `Execute` `Batch`, I want query all slow operations in one query.

**Comment by quaff** [comment]
> > duration is for the selected service/operation, it applies anywhere in the trace
> 
> There are many operations including `Query` `Update` `Execute` `Batch`, I want query all slow operations in one query.

@yurishkuro Any thoughts?

File: src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java [hunk 1]
```diff
@@ -33,6 +34,13 @@
    */
   static final StringTag PEER_ADDRESS = new StringTag("peer.address");
 
+  static final BooleanTag SLOW = new BooleanTag("slow");
+
+  /**
+   * can be modified by application code
+   */
+  public static int slowQueryThresholdMs = Integer.getInteger("io.opentracing.contrib.jdbc.slowQueryThresholdMs", 100);
```
Comment (yurishkuro) [inline src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java:42]: For backwards compatibility, I would use `0` as the default value, and skip bypass the new logic if slowQueryThresholdMs. This way people who want the feature can enable it with non-zero value, while others won't be paying the perf penalty of getting two more timers, etc.

Comment (quaff) [reply src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java]: Updated, WDYT @malafeev ?

File: src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java [hunk 2]
```diff
@@ -68,14 +76,18 @@ static Span buildSpan(String operationName,
...
-       JdbcTracingUtils.onError(e, span);
-       throw e;
-     } finally {
-       span.finish();
-     }
+    } catch (Exception e) {
+      JdbcTracingUtils.onError(e, span);
+      throw e;
+    } finally {
+      if (System.nanoTime() - time > TimeUnit.MILLISECONDS.toNanos(slowQueryThresholdMs)) {
```
Comment (yurishkuro) [inline src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java:86]:
```suggestion
      if (slowQueryThresholdMs  > 0 && System.nanoTime() - time > TimeUnit.MILLISECONDS.toNanos(slowQueryThresholdMs)) {
```

File: src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java [hunk 3]
```diff
@@ -68,14 +76,18 @@ static Span buildSpan(String operationName,
 
     final Span span = buildSpan(operationName, sql, connectionInfo, withActiveSpanOnly,
     ignoreStatements, tracer);
-     try (Scope ignored = tracer.activateSpan(span)) {
+    long time = System.nanoTime();
```
Comment (yurishkuro) [inline src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java:79]:
```suggestion
    long time = slowQueryThresholdMs  > 0 ? System.nanoTime() : 0;
```
don't obtain time if not necessary

File: src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java [hunk 4]
```diff
@@ -85,21 +97,25 @@ static Span buildSpan(String operationName,
...
- } catch (Exception e) {
-   JdbcTracingUtils.onError(e, span);
-   throw e;
- } finally {
-   span.finish();
- }
-}
+    final Span span = buildSpan(operationName, sql, connectionInfo, withActiveSpanOnly,
+        ignoreStatements, tracer);
+    long time = System.nanoTime();
```
Comment (yurishkuro) [inline src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java:106]:
```suggestion
    long time = slowQueryThresholdMs  > 0 ? System.nanoTime() : 0;
```

**Comment by quaff** [comment]
@yurishkuro updated, thanks.

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**